src/db/MariadbIfc.py

```diff
@@ -0,0 +1,504 @@
+        cmd        = ""
+        result     = None
+        pic_id     = 0
+
+        if picture_id == None:
+
+            self.db_log.info(f"Getting picture ID")
+
+            cmd = (self.prof_cmds['get_profile']) % (profile_id)
+            self.db_log.debug(f"Executing get profile command {cmd}")
+            cursor.execute(cmd)
+
+            profile = cursor.fetchone()
+
+            if profile == None:
+
+                self.db_log.warn(f"Could not find the profile {profile_id} in the DB!")
+                return ""
+
+        self.db_log.info(f"Getting picture.")
+        cmd = (self.pict_cmds['get_image']) % (picture_id if picture_id != None else profile[int(self.prof_cmds['pic_id_index'])])
+        self.db_log.debug(f"Executing get picture command {cmd}")
+        cursor.execute(cmd)
+        #The cursor object doesn't appear to actually provide a better way
+        #to determine if the cursor has a result.
+        img = cursor.fetchone()
+
+        if (img == None):
+
+            self.db_log.warning(f"Picture ID {profile[int(self.prof_cmds['pic_id_index'])]} not found!")
+            return ""
+
+        else:
+
+            self.db_log.debug(f"Got picture: {img[0]}")
+            return img[0]
+
+    def GetProfile(self,
+                   id         : Optional[str] = "ffffffff-ffff-ffff-ffff-fffffffffffe") -> Optional[pg.Profile]:
+        """Returns a given profile for a given user.
+
+            Input: self - Pointer to the current object instance.
+                   id - The profile to get, defaults to the test profile.
+
+            Output: str - The profile object found by the search, if any.
+        """
+        cursor     = self.con.cursor(buffered=False)
+        cmd        = ""
+        profile    = None
+        result     = None
+
+        self.db_log.info(f"Getting profile")
+        cursor.execute((self.prof_cmds['get_profile']) % id)
+        #The cursor object doesn't appear to actually provide a better way
+        #to determine if the cursor has a result.
+        result = cursor.fetchone()
+
+        if (result == None):
+
+            self.db_log.warning(f"Profile not found!: {id}")
+
+        else:
+
+            self.db_log.debug(f"Got profile: {result}")
+            #Since profiles are stored as individual elements in the DB for
+            #statistical analysis, the actual profile object needs to be
+            #re-created before it can be returned (since the rest of the code
+            #expects to interact with an object).
+            stats=sc.Stats(rarity=rc.RarityList(int(result[21])),
+                           agility=result[5],
+                           defense=result[6],
+                           endurance=result[7],
+                           luck=result[8],
+                           strength=result[9])
+            self.db_log.debug(f"Made stats: {pic.dumps(stats)}")
+             #TODO: find a way to better map these.  Blah blah factory.
```

Comment:
Could give it the profile object instead of each individual argument?

Reply:
It's just that initializing a profile object is taking like 20+ arguments. Nothing to really worry about. Was maybe thinking that maybe just passing the connection object would make it cleaner and then inside the init you could do all the assignments.